worker/pkg/workflows/datasync/activities/activities.go

```diff
@@ -544,57 +543,66 @@ func getMysqlDsn(
 	}
 }
 
-func (b *benthosBuilder) buildProcessorConfig(ctx context.Context, cols []*mgmtv1alpha1.JobMapping) (*neosync_benthos.ProcessorConfig, error) {
+func (b *benthosBuilder) buildProcessorConfigs(ctx context.Context, cols []*mgmtv1alpha1.JobMapping) ([]*neosync_benthos.ProcessorConfig, error) {
+	var jsFunctions, benthosOutputs []string
+	var err error
+
+	jsFunctions, benthosOutputs = b.extractJsFunctionsAndOutputs(cols, jsFunctions, benthosOutputs)
```

Comment:
Why is this not just:
```
jsFunctions, benthosOutputs = b.extractJsFunctionsAndOutputs(cols)
```

Or even better:

```
jsCode := b.buildJsCodeConfig(cols)
```

This reduces this function to:

```
mutation, err := b.buildMutationConfig(ctx, cols)
if err != nil {
  return err
}
jsCode, err := b.buildJsCodeConfig(ctx, cols)
if err != nil {
  return err
}

processorConfigs := []*neosync_benthos.ProcessorConfig{}
if len(mutation) > 0 {
  processorConfigs = append(processorConfigs, &neosync_benthos.ProcessorConfig{Mutation: &mutation})
}
if len(jsCode) > 0 {
  processorConfigs = append(processorConfigs, &neosync_benthos.ProcessorConfig{Javascript: &JavascriptConfig{Code: jsCode}})
}
return processorConfigs
```

Reply:
@nickzelei yeah i suppose can we do that too :) updated